src/pint/residuals.py

```diff
@@ -123,6 +125,26 @@ def resids_value(self):
         else:
             return self.resids.to_value(self.unit)
 
+    @property
+    def dof(self):
+        """Return number of degrees of freedom for the model."""
+        if self._is_combined:
+            raise RuntimeError(
+                "Please use the `.dof` in the CombinedResidual"
+                " class. The individual residual's dof is not "
+                "calculated correctly in the combined residuals."
+            )
+        dof = self.toas.ntoas
+        for cp in self.model.components.values():
+            if "delay" in cp.modeled_quantity or "phase" in cp.modeled_quantity:
```

Comment:
What is `modeled_quantity` for? This ignores model components that affect DM, which do affect timing (see all our narrowband work). Do you have test cases that check the `.dof` value for phase-only residuals where the model includes DM fitting? What is the right way to compute a `.dof` for a model that includes `EQUAD`, `EFAC`, or `ECORR`?